featurebyte/query_graph/graph.py

```diff
@@ -307,6 +309,39 @@ def extract_operation_structure(
         )
         return op_struct_info.operation_structure_map[node.name]
 
+    def extract_table_id_to_table_column_names(self, node: Node) -> Dict[ObjectId, Set[str]]:
+        """
+        Extract table ID to table column names based on the graph & given target node.
+        This mapping is used to prune the view graph node parameters or extract the table columns used by the
+        query graph object.
+
+        Parameters
+        ----------
+        node: Node
+            Target node used to the mapping
+
+        Returns
+        -------
+        dict[ObjectId, set[str]]
+            Table ID to table column names
+        """
+        operation_structure = self.extract_operation_structure(node, keep_all_source_columns=True)
+        # prepare table ID to source column names mapping, use this mapping to prune the view graph node parameters
+        table_id_to_source_column_names: Dict[ObjectId, Set[str]] = defaultdict(set)
+        for col in operation_structure.columns:
```

Comment:
It seems this kind of iteration appears also in `iterate_source_columns_or_aggregations`. Might be useful to extract a helper function for it.